components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1548,6 +1548,59 @@ END SUBROUTINE access_lookup_table_coll
 
   !==========================================================================================!
 
+  real(rtype) function svp_murphy_koop(t, i_type)
+
+    implicit none
+
+    !-------------------------------------------------------------------
+    ! Compute saturation vapor pressure (returned in units of pa)
+    ! Inputs:
+    ! "t", units [K]
+    !  i_type refers to saturation with respect to liquid (0) or ice (1)
+    !--------------------------------------------------------------------
+
+    !Murphy & Koop (2005)
+    real(rtype), intent(in) :: t
+    integer,intent(in)      :: i_type
+
+    !local vars
+    character(len=1000) :: err_msg
+    real(rtype)         :: logt
+
+    !parameters for ice saturation eqn
+    real(rtype), parameter :: ic(4)  =(/9.550426_rtype, 5723.265_rtype, 3.53068_rtype, &
+         0.00728332_rtype/)
+
+    !parameters for liq saturation eqn
+    real(rtype), parameter :: lq(10) = (/54.842763_rtype, 6763.22_rtype, 4.210_rtype, &
+         0.000367_rtype, 0.0415_rtype, 218.8_rtype, 53.878_rtype, 1331.22_rtype,       &
+         9.44523_rtype, 0.014025_rtype /)
+
```

Comment:
Since we keep getting weird failures out of polysvp, maybe it would make sense to check if t<0 and t=NaN and error usefully before launching into calculations?

Reply:
I wrote a function to check temperature and called it from qv_sat routine.